cdci_data_analysis/analysis/email_helper.py

```diff
@@ -353,25 +353,60 @@ def store_email_api_code_attachment(api_code, status, scratch_dir, sending_time=
     return attachment_file_path
 
 
-def is_email_to_send_run_query(logger, status, time_original_request, scratch_dir, job_id, config, decoded_token=None):
+def log_email_sending_info(logger, status, time_request, scratch_dir, job_id, additional_info_obj=None):
+    path_email_history_folder = os.path.join(scratch_dir, 'email_history')
+    if not os.path.exists(path_email_history_folder):
+        os.makedirs(path_email_history_folder)
+    history_info_obj = dict(time=timestamp2isot(time_request),
+                            status=status,
+                            job_id=job_id)
+    if additional_info_obj is not None:
+        history_info_obj['additional_information'] = additional_info_obj
+    email_history_log_fn = os.path.join(path_email_history_folder, 'email_history_log.log')
+    email_history_log_obj_list = []
+    # Opening JSON list file
+    if os.path.exists(email_history_log_fn):
+        with open(email_history_log_fn, 'r') as openfile:
+            email_history_log_obj_list = json.load(openfile)
+    # append new log object
+    email_history_log_obj_list.append(history_info_obj)
+    with open(email_history_log_fn, 'w') as outfile:
```

Comment:
will this happen with every `run_query`?

Reply:
yes, at every call of the `log_email_sending_info` method

Reply:
It creates not quite as much load as callback, but still a lot. Could you restrict this to log, unless the email is actually submitted?

Reply:
now, in the case of `run_query`, it logs in case the email cannot be sent

> unless the email is actually submitted?

you mean sent ? so to log in case the email sending fails?

Reply:
> now, in the case of `run_query`, it logs in case the email cannot be sent
> 
> > unless the email is actually submitted?
> 
> you mean sent ? so to log in case the email sending fails?

I really mean when it's sent. 
Because the problem now is that several near-identical emails are produced and sent for the same event.
When fails - we get a sentry. Nothing new should be done in this case.